diff --git a/src/type.hh b/src/type.hh
index c5e5071..b6f1b79 100644
--- a/src/type.hh
+++ b/src/type.hh
@@ -516,6 +516,12 @@ protected:
 public:
   TypeStruct(const TypeStruct &op);	///< Construct from another TypeStruct
   TypeStruct(void) : Datatype(0,-1,TYPE_STRUCT) { flags |= type_incomplete; }	///< Construct incomplete/empty TypeStruct
+  TypeStruct(const std::string& nm) : Datatype(0,-1,TYPE_STRUCT) {	///< Construct incomplete TypeStruct with a name
+    name = nm;
+    displayName = nm;
+    flags |= type_incomplete;
+    id = hashName(nm);
+  }
   vector<TypeField>::const_iterator beginField(void) const { return field.begin(); }	///< Beginning of fields
   vector<TypeField>::const_iterator endField(void) const { return field.end(); }	///< End of fields
   virtual const TypeField *findTruncation(int8 off,int4 sz,const PcodeOp *op,int4 slot,int8 &newoff) const;
@@ -549,6 +555,12 @@ protected:
 public:
   TypeUnion(const TypeUnion &op);	///< Construct from another TypeUnion
   TypeUnion(void) : Datatype(0,-1,TYPE_UNION) { flags |= (type_incomplete | needs_resolution); }	///< Construct incomplete TypeUnion
+  TypeUnion(const std::string& nm) : Datatype(0,-1,TYPE_UNION) {	///< Construct incomplete TypeUnion with a name
+    name = nm;
+    displayName = nm;
+    flags |= type_incomplete | needs_resolution;
+    id = hashName(nm);
+  }
   const TypeField *getField(int4 i) const { return &field[i]; }	///< Get the i-th field of the union
   virtual const TypeField *findTruncation(int8 offset,int4 sz,const PcodeOp *op,int4 slot,int8 &newoff) const;
   //  virtual Datatype *getSubType(int8 off,int8 *newoff) const;
